sys/compat/linprocfs/linprocfs.c

```diff
@@ -2092,6 +2095,176 @@ linprocfs_domax_map_cnt(PFS_FILL_ARGS)
 	return (0);
 }
 
+/*
+ * Filler function for proc/sysvipc/msg
+ */
+static int
+linprocfs_dosysvipc_msg(PFS_FILL_ARGS)
+{
+	struct msqid_kernel *msqids;
+	u_long id, msgmni;
+	size_t sz;
+	int error;
+
+	sbuf_printf(sb,
+	    "%10s %10s %4s  %10s %10s %5s %5s %5s %5s %5s %5s %10s %10s %10s\n",
+	    "key", "msqid", "perms", "cbytes", "qnum", "lspid", "lrpid",
+	    "uid", "gid", "cuid", "cgid", "stime", "rtime", "ctime");
+
+again:
+	msgmni = msginfo.msgmni;
+	sz = sizeof(struct msqid_kernel) * msgmni;
+	msqids = malloc(sz, M_TEMP, M_NOWAIT);
+	if (msqids == NULL)
+		return (ENOMEM);
+	if (msgmni != msginfo.msgmni) {
+		free(msqids, M_TEMP);
+		goto again;
+	}
```

Comment:
Why is msginfo.msgmni changing? It would be better to just reply on the sysctl to get the data. It will return ENOMEM when the data size is too small. The case where it's too big is now handled corrected.